src/compiler/visitorPublic.ts

```diff
@@ -164,12 +221,33 @@ export function visitArray<T extends Node, U extends T>(nodes: readonly T[] | un
         count = length - start;
     }
 
-    return visitArrayWorker(nodes, visitor, test, start, count) as readonly U[];
+    return visitArrayWorker(nodes, visitor, test, start, count);
 }
 
 /** @internal */
-function visitArrayWorker<T extends Node>(nodes: readonly T[], visitor: Visitor, test: ((node: Node) => boolean) | undefined, start: number, count: number): readonly T[] | undefined {
-    let updated: T[] | undefined;
+export function visitArrayWorker<TIn extends Node, TInArray extends readonly TIn[], TOut extends Node>(

```

Comment:
Why is this now exported? It shouldn't be called by anything outside of this file.